lib/Connection.js

```diff
@@ -71,7 +71,7 @@ class Connection extends EventEmitter {
     debugProtocol('SEND ► ' + message);
     this._ws.send(message);
     return new Promise((resolve, reject) => {
-      this._callbacks.set(id, {resolve, reject, method});
+      this._callbacks.set(id, {resolve, reject: rewriteError.bind(null, new Error(), reject), method});
```

Comment:
instead of binding in the `rewriteError` function, let's just save the error as a part of the already-stored `resolve` and `reject` methods.

Reply:
Done.